Add support for go.formatTool: "goformat" #1656

mbenkmann opened the pull request:
See issue #1656

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vscode-go?pullRequest=1670) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vscode-go?pullRequest=1670) before we can accept your contribution.<br/><br/>:x: mbenkmann [sign now](https://cla.opensource.microsoft.com/Microsoft/vscode-go?pullRequest=1670)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vscode-go?pullRequest=1670) it.</sub>

ramya-rao-a commented on src/goInstallTools.ts:
```diff
@@ -33,6 +33,7 @@ const allTools: { [key: string]: string } = {
 	'gogetdoc': 'github.com/zmb3/gogetdoc',
 	'goimports': 'golang.org/x/tools/cmd/goimports',
 	'goreturns': 'github.com/sqs/goreturns',
+	'goformat': 'github.com/mbenkmann/goformat/goformat',
```
@mbenkmann When I run `go get github.com/mbenkmann/goformat/goformat`, I get the error "package github.com/mbenkmann/goformat/goformat: cannot find package "github.com/mbenkmann/goformat/goformat" in any of:..."

Does it expect that the user has already cloned the repo?

mbenkmann commented:
I verified that the installation worked from VSCode by removing the package from my GOPATH. There were no issues.
I just tried it again with a blank GOPATH and it worked.

    mkdir /tmp/go
    export GOPATH=/tmp/go
    go get github.com/mbenkmann/goformat/goformat

mbenkmann commented:
I think I know what the problem is. I have symlinks in the repo to make it buildable outside of GOPATH (for people who like to download stuff in /tmp). I assume you are on Windows and those symlinks don't clone for you, right?

mbenkmann commented:
Okay. I've fixed the repository. It should work without symlinks now. I've also changed the URL in goInstallTools.ts to the canonical URL which uses my own domain, because I've noticed that go get github/... creates 2 copies of the source tree, one in src/github.com/goformat and one in src/winterdrache.de/goformat
